packages/good-design/src/apps/ubi/context/ClaimContext.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { createContext, FC, PropsWithChildren, useContext, useCallback, useEffect, useState } from "react";
+import {
+  getRecentClaims,
+  SupportedChains,
+  useClaim,
+  useClaimedAlt,
+  useGetMemberUBIPools,
+  useMultiClaim
+} from "@gooddollar/web3sdk-v2";
+import { QueryParams, useEthers } from "@usedapp/core";
+import { isArray, isEmpty, isUndefined, noop } from "lodash";
+import { Lock } from "async-await-mutex-lock";
+
+import { getUnclaimedPools } from "../utils/pools";
+import { ClaimContextProps } from "../types";
+import { useFormatClaimTransactions } from "../../../hooks";
+
+const ClaimContext = createContext<ClaimContextProps | undefined>(undefined);
+
+export const useClaimContext = () => {
+  const context = useContext(ClaimContext);
+  if (!context) {
+    throw new Error("useClaimContext must be used within a ClaimProvider");
+  }
+  return context;
+};
+
+const explorerPollLock = new Lock();
+
+export const ClaimProvider: FC<
+  {
+    explorerEndPoints: { [key in keyof typeof SupportedChains]: string };
```

Comment:
why is that an input? it should at least have default values

Reply:
to support different endpoints (if needed) for either wallet/dapp.

Reply:
I don't know if default values make sense if only fuse explorer works without setting an api-key?

Reply:
explorer.celo.org also works without a key